### PR TITLE
add typings for HTTPS agent for fetch and Request

### PR DESCRIPTION
**Pull request by cliffordfajardo**
### What is the change?
- Add typings for `https` agent
- Add correct typings fetch's `options` param

### Why are we making this change?
- Allow a user to pass `https` to `Request
- Allow agent & other [`RequestExtraOptions` values](https://github.com/remix-run/web-std-io/blob/main/packages/fetch/src/fetch.js#L184) to be passed to `fetch`

### Tests
Type changes only. Ran npm test inside of `package/fetch` and all tests pass

<details>
<summary>Test output for  `package/fetch` </summary>

```
 247 passing (4s)
  4 pending

-----------------|---------|----------|---------|---------|-----------------------------------------
File             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                       
-----------------|---------|----------|---------|---------|-----------------------------------------
All files        |   97.15 |    96.79 |   89.32 |   97.15 |                                         
 src             |   96.69 |    96.58 |   86.58 |   96.69 |                                         
  body.js        |   96.97 |    95.45 |   97.36 |   96.97 | 211-212,250-254,263,464-473             
  fetch.js       |   99.46 |    96.38 |     100 |   99.46 | 117-118                                 
  headers.js     |   92.73 |    93.75 |   81.81 |   92.73 | 16-24,29-38,95-96                       
  lib.node.js    |     100 |      100 |     100 |     100 |                                         
  package.js     |     100 |      100 |     100 |     100 |                                         
  request.js     |   95.62 |      100 |   56.25 |   95.62 | ...-170,173-174,178-179,183-184,188-189 
  response.js    |   98.59 |      100 |    90.9 |   98.59 | 59-60                                   
 src/errors      |     100 |      100 |     100 |     100 |                                         
  abort-error.js |     100 |      100 |     100 |     100 |                                         
  base.js        |     100 |      100 |     100 |     100 |                                         
  fetch-error.js |     100 |      100 |     100 |     100 |                                         
 src/utils       |   99.27 |     97.4 |     100 |   99.27 |                                         
  form-data.js   |   98.29 |       92 |     100 |   98.29 | 115-116                                 
  get-search.js  |     100 |      100 |     100 |     100 |                                         
  is-redirect.js |     100 |      100 |     100 |     100 |                                         
  is.js          |     100 |      100 |     100 |     100 |                                         
  utf8.js        |     100 |      100 |     100 |     100 |                                         
-----------------|---------|----------|---------|---------|-----------------------------------------
```
<details>

### CODE REVIEW

**Comment by cliffordfajardo** [comment]
@mcansh - trying to add a change set but I only see the option to make a `major` bump in the CLI 🤔 
Is that correct?

CleanShot 2022-09-20 at 13 54 00@2x

**Comment by mcansh** [comment]
> @mcansh - trying to add a change set but I only see the option to make a `major` bump in the CLI 🤔 
> 
> Is that correct?
> 
> 
> 
> CleanShot 2022-09-20 at 13 54 00@2x
> 
> 

if you hit enter without selecting a package, it should prompt for minor and then patch. however you can always change the changeset after :)

however, i usually just do it via the changeset bot prompt

**Comment by cliffordfajardo** [comment]
Updated to patch in a following commit 😄 
Hopefully github squash & merge should fix the unnecessary commit

EDIT: updating to `minor`

**Comment by cliffordfajardo** [comment]
Updated

**Comment by cliffordfajardo** [comment]
@mcansh - any other changes you think I should make?

**Comment by mcansh** [comment]
> @mcansh - any other changes you think I should make?

sorry @cliffordfajardo i was out the last 2 weeks, but was popping in and out of GitHub, I'll get this reviewed shortly :)

**Review by mcansh — APPROVED** [review]
LGTM, gonna get another approval real quick before merging :)

**Review by jacob-ebey — APPROVED** [review]

**Pull request merged**
